client-js/src/client/actor-request-factory.js

```diff
@@ -281,27 +281,38 @@ class Targets {
   }
 }
 
-const INVALID_FILTER_TYPE =
-  'All filters passed to QueryFilter#where() must be of a single type: ' +
-  'either ColumnFilter or CompositeColumnFilter.';
-
 /**
- * A builder for creating `Query` instances. A more flexible approach to query creation
- * than using a `QueryFactory`.
+ * An abstract base for builders that create `Message` instances which have a `Target`
+ * and a `FieldMask` as attributes.
+ *
+ * <p>The `Target` matching the builder configuration is accessed with `#getTarget()`,
+ * while the `FieldMask` is retrieved with `#getMask()`.
+ *
+ * This classes public API is inspired by the SQL syntax.
+ * ```javascript
+ *     select(CUSTOMER_TYPE) // returning <AbstractTargetBuilder> instance
```

Comment:
I don't understand how the instance of abstract type is returned. I would expect that an instance of descendant is returned.

Let's discuss.